lib/esConnector.js

```diff
@@ -1266,6 +1276,309 @@ ESConnector.prototype.updateOrCreate = function updateOrCreate(modelName, data,
 };
 
 /**
+ * direct access to the Elasticsearch bulk API
+ * https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html
+ *
+ * @param modelName {String} - required, model that is accociated with this connector
+ * @param params {Object} - required, same as the native bulk function uses
+ * @param done {Function} - allows async execution
+ * @throws {Error} - if argument rules are violated
+ * @returns {Promise} OR {undefined} if callback (done) is provided
+
+ */
+ESConnector.prototype.bulk = function bulk(modelName, params, done){
+  var self = this;
+  if (self.debug) {
+    log('ESConnector.prototype.bulk', modelName, params);
+  }
+
+  if (typeof params !== 'object' || params === null) {
+    throw new Error('invalid argument: params must be defined');
+  }
+  if (Array.isArray(params.body) === false) {
+    throw new Error('invalid argument: params.body must be an array');
+  }
+  if (params.body.length <= 0) {
+    var res = {took: 0, errors: false, items: []};
+    if (typeof done === 'function') {
+      return done(null, res);
+    } else {
+      return Promise.resolve(res);
+    }
+  }
+
+  return self.db
+    .bulk(params)
+    .then(
+      function(res){
+        log('ESConnector.prototype.bulk', 'response', res);
+        if (typeof done === 'function') {
+          log('ESConnector.prototype.bulk', 'will invoke callback with full response:', res);
+          done(null, res);
+        } else {
+          return Promise.resolve(res);
+        }
+      },
+      function(err){
+        console.trace(err.message);
+        if (typeof err !== 'undefined' && err !== null) {
+          if (typeof done === 'function') {
+            done(err, null);
+          } else {
+            return Promise.reject(err);
+          }
+        }
```

Comment:
How are you planning to handle if indeed error was undefined or null? Or does it even enter this function block if there's no error?